### DISCUSSION TITLE
Are MultiPolyline and MultiPolygon planned to be added in the future?

### DISCUSSION THREAD

**Discussion by MatiasHiltunen** [fleaflet/flutter_map#1025] [Ideas]
My question comes from issue where perfomance is going to be a problem, I have to render thousands of Polylines and Polygons. With about 3000 Polylines the jank is unbearable. With Leaflet js in web I have solved that issue with MultiPolylines and it works smoothly.

**Comment by JaffaKetchup** [comment]
Sorry, not sure what a `MultiPolyline` is and how it works? Can you describe it? It may be possible that you or someone else could make it into a plugin.

**Reply by MatiasHiltunen** [reply]
Sure, from leaflet's js documentation:
// Polyline
var latlngs = [
    [45.51, -122.68],
    [37.77, -122.43],
    [34.04, -118.2]
];

// MultiPolyline 
var latlngs = [
    [[45.51, -122.68],
     [37.77, -122.43],
     [34.04, -118.2]],
    [[40.78, -73.91],
     [41.83, -87.62],
     [32.76, -96.72]]
];

// Polygon
var latlngs = [
  [[37, -109.05],[41, -109.03],[41, -102.05],[37, -102.04]], // outer ring
  [[37.29, -108.58],[40.71, -108.58],[40.71, -102.50],[37.29, -102.50]] // hole
];


// MultiPolygon 

var latlngs = [
  [ // first polygon
    [[37, -109.05],[41, -109.03],[41, -102.05],[37, -102.04]], // outer ring
    [[37.29, -108.58],[40.71, -108.58],[40.71, -102.50],[37.29, -102.50]] // hole
  ],
  [ // second polygon
    [[41, -111.03],[45, -111.04],[45, -104.05],[41, -104.05]]
  ]
];

**Reply by JaffaKetchup** [reply]
Ok, I think this can already be done (sorry I'm misunderstanding - it's likely)? Using polygons as an example, the `PolygonLayerOptions` takes a `List<Polygon>`. Does leaflet.js use a different method to render `MultiPolygon`s to improve efficiency?

**Reply by MatiasHiltunen** [reply]
I think that is the case, List<Polygon> is filled with (in my case) thousands of Polygon instances as with MultiPolygon there would be only one instance which renders the polygons on map

**Reply by JaffaKetchup** [reply]
Not sure, I've only used leaflet.js a few times, but could you use the holes property in a Polygon to separate one huge polygon, making it look visually like one? Unfortunately I cannot help any more that this.

**Reply by MatiasHiltunen** [reply]
I will let you know if I manage to solve the problem. First I will try to modify the Polygon/Polyline so that it will take a multidimensional List of Polygons/Polylines

**Reply by MatiasHiltunen** [reply]
Okay, now that I checked how the things are done under the hood I don't think my previous idea would do anything to make this work better. I will keep investigating this more, this can also be issue with my code in map state as I use provider to feed the List<Polygon> to the Polygon options

**Comment by ibrierley** [comment]
If you can batch all the paths into one super path (path.addPath()) it may
well help.
